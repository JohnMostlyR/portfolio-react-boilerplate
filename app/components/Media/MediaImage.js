import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { alignmentToFlex } from './helper';

const StyledImg = styled.img.attrs({
  src: props => props.imageSource || '',
  alt: props => props.imageAlt || '',
})`
  align-self: ${props => alignmentToFlex(props.imageAlign)};
  height: ${props => props.imageHeight};
  margin: ${props => (props.reverse ? '0 0 0 0.5em' : '0 0.5em 0 0')};
  order: ${props => (props.reverse ? 2 : 1)};
  width: ${props => props.imageWidth};
  max-width: 100%;
`;

const MediaImage = props => <StyledImg {...props} />;

MediaImage.propTypes = {
  imageAlign: PropTypes.oneOf(['top', 'middle', 'bottom']),
  imageAlt: PropTypes.string.isRequired,
  imageHeight: PropTypes.string,
  imageLink: PropTypes.string,
  imageSource: PropTypes.string.isRequired,
  imageWidth: PropTypes.string,
  reverse: PropTypes.bool,
};

MediaImage.defaultProps = {
  imageAlign: 'top',
  imageHeight: 'auto',
  imageWidth: '100%',
  reverse: false,
};

export default MediaImage;
