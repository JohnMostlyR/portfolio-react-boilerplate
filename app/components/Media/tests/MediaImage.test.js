import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import MediaImage from '../MediaImage';

describe('<MediaImage />', () => {
  const src = '/test.png';
  const alt = 'test';

  it('should render and match the snapshot', () => {
    const wrapper = renderer.create(
      <MediaImage imageSource={src} imageAlt={alt} />,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render an <img /> element', () => {
    const wrapper = mount(<MediaImage imageSource={src} imageAlt={alt} />);
    expect(wrapper.containsMatchingElement(<img src={src} alt={alt} />)).toBe(
      true,
    );
    wrapper.unmount();
  });

  describe('should adopt the "imageAlign" prop', () => {
    let imageAlign = '';
    let wrapper = null;

    it('set to "top"', () => {
      imageAlign = 'top';
      wrapper = shallow(
        <MediaImage imageSource={src} imageAlt={alt} imageAlign={imageAlign} />,
      );
      expect(wrapper.prop('imageAlign')).toBe(imageAlign);
      expect(wrapper).toMatchSnapshot();
    });

    it('set to "middle"', () => {
      imageAlign = 'middle';
      wrapper = shallow(
        <MediaImage imageSource={src} imageAlt={alt} imageAlign={imageAlign} />,
      );
      expect(wrapper.prop('imageAlign')).toBe(imageAlign);
      expect(wrapper).toMatchSnapshot();
    });

    it('set to "bottom"', () => {
      imageAlign = 'bottom';
      wrapper = shallow(
        <MediaImage imageSource={src} imageAlt={alt} imageAlign={imageAlign} />,
      );
      expect(wrapper.prop('imageAlign')).toBe(imageAlign);
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should adopt the "imageHeight" prop', () => {
    const imageHeight = '10px';
    const wrapper = shallow(
      <MediaImage
        imageSource="/test.png"
        imageAlt="test"
        imageHeight={imageHeight}
      />,
    );
    expect(wrapper.prop('imageHeight')).toBe(imageHeight);
    expect(wrapper).toMatchSnapshot();
  });

  it('should adopt the "imageWidth" prop', () => {
    const imageWidth = '10px';
    const wrapper = shallow(
      <MediaImage
        imageSource="/test.png"
        imageAlt="test"
        imageWidth={imageWidth}
      />,
    );
    expect(wrapper.prop('imageWidth')).toBe(imageWidth);
    expect(wrapper).toMatchSnapshot();
  });

  it('should adopt the "reverse" prop', () => {
    const reverse = false;
    const wrapper = shallow(
      <MediaImage imageSource="/test.png" imageAlt="test" reverse={reverse} />,
    );
    expect(wrapper.prop('reverse')).toBe(reverse);
    expect(wrapper).toMatchSnapshot();
  });
});
