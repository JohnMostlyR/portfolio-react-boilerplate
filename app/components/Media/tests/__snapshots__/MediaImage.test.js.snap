// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaImage /> should adopt the "imageAlign" prop set to "bottom" 1`] = `
<MediaImage__StyledImg
  imageAlign="bottom"
  imageAlt="test"
  imageHeight="auto"
  imageSource="/test.png"
  imageWidth="100%"
  reverse={false}
/>
`;

exports[`<MediaImage /> should adopt the "imageAlign" prop set to "middle" 1`] = `
<MediaImage__StyledImg
  imageAlign="middle"
  imageAlt="test"
  imageHeight="auto"
  imageSource="/test.png"
  imageWidth="100%"
  reverse={false}
/>
`;

exports[`<MediaImage /> should adopt the "imageAlign" prop set to "top" 1`] = `
<MediaImage__StyledImg
  imageAlign="top"
  imageAlt="test"
  imageHeight="auto"
  imageSource="/test.png"
  imageWidth="100%"
  reverse={false}
/>
`;

exports[`<MediaImage /> should adopt the "imageHeight" prop 1`] = `
<MediaImage__StyledImg
  imageAlign="top"
  imageAlt="test"
  imageHeight="10px"
  imageSource="/test.png"
  imageWidth="100%"
  reverse={false}
/>
`;

exports[`<MediaImage /> should adopt the "imageWidth" prop 1`] = `
<MediaImage__StyledImg
  imageAlign="top"
  imageAlt="test"
  imageHeight="auto"
  imageSource="/test.png"
  imageWidth="10px"
  reverse={false}
/>
`;

exports[`<MediaImage /> should adopt the "reverse" prop 1`] = `
<MediaImage__StyledImg
  imageAlign="top"
  imageAlt="test"
  imageHeight="auto"
  imageSource="/test.png"
  imageWidth="100%"
  reverse={false}
/>
`;

exports[`<MediaImage /> should render and match the snapshot 1`] = `
<img
  alt="test"
  className="MediaImage__StyledImg-eVnySD xBBKk"
  src="/test.png"
/>
`;
